lib/add_esm_snapshot_auth.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from uaclient import config, defaults, log
+from uaclient.api.u.pro.status.is_attached.v1 import _is_attached
+from uaclient.entitlements.entitlement_status import ApplicationStatus
+from uaclient.entitlements.esm import ESMAppsEntitlement, ESMInfraEntitlement
+
+LOG = logging.getLogger("ubuntupro.lib.add_esm_snapshot_auth")
+
+ESM_APPS_SNAPSHOT_URLS = [
+    "snapshot.apps-security.esm.ubuntu.com/apps/ubuntu/",
+    "snapshot.apps-updates.esm.ubuntu.com/apps/ubuntu/",
+]
+ESM_INFRA_SNAPSHOT_URLS = [
+    "snapshot.infra-security.esm.ubuntu.com/infra/ubuntu/",
+    "snapshot.infra-updates.esm.ubuntu.com/infra/ubuntu/",
+]
+
+
+def add_esm_snapshot_auth(cfg):
+    if not _is_attached(cfg).is_attached:
+        LOG.info("Not attached. Ending.")
```

Comment:
Can we just log that we are not adding the `esm snapshot` url here ?

Reply:
Yep good call